pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -417,15 +442,38 @@ func getObjectVersion(id string) string {
 }
 
 func isSecretDisabledError(err error) bool {
-	var de autorest.DetailedError
-	if errors.As(err, &de) {
-		var re *azure.RequestError
-		if errors.As(de.Original, &re) {
-			if re.ServiceError.Code == "SecretDisabled" {
-				return true
+	// AzureError defines the structure of the error response from Azure Key Vault
+	type AzureError struct {
```

Comment:
Seems this is defined in terms of the KeyVaultError from KV reference: https://learn.microsoft.com/en-us/rest/api/keyvault/keys/get-keys/get-keys?view=rest-keyvault-keys-7.4&tabs=HTTP#keyvaulterror
We can add a link to it